packages/contractkit/src/wrappers/Accounts.ts

```diff
@@ -317,8 +313,42 @@ export class AccountsWrapper extends BaseWrapper<Accounts> {
    * @param name A string to set as the name of the account
    * @param dataEncryptionKey secp256k1 public key for data encryption. Preferably compressed.
    * @param walletAddress The wallet address to set for the account
+   * @param
    */
-  setAccount = proxySend(this.kit, this.contract.methods.setAccount)
+  setAccount(
+    name: string,
+    dataEncryptionKey: string,
+    walletAddress: Address,
+    proofOfPossession: Signature | null = null
+  ): CeloTransactionObject<void> {
+    if (proofOfPossession) {
+      return toTransactionObject(
+        this.kit,
+        this.contract.methods.setAccount(
+          name,
+          // @ts-ignore
+          dataEncryptionKey,
+          walletAddress,
+          proofOfPossession.v,
+          proofOfPossession.r,
+          proofOfPossession.s
+        )
+      )
+    } else {
+      return toTransactionObject(
+        this.kit,
+        this.contract.methods.setAccount(
```

Comment:
Why can we call `setAccount` here without a proof of possession?

Reply:
We cannot, and the tests failed appropriately

Reply:
I rely on the estimateGas failing here as there is no easy access to `msg.sender`